tools/vm-migrator/instance.py

```diff
@@ -0,0 +1,397 @@
+                if (not alias_ip_name):
+                    alias_ip_name = name + "-alias-ip-" + str(i)
+
+                # Passing None in actual ip will reserve a random ip for the name
+                logging.info("reserving alias ip %s for machine %s" %
+                             (actual_ip, name))
+                reserve_internal_ip(compute, project, alias_ip_name,
+                                    get_region_from_zone(zone), subnet,
+                                    actual_ip)
+                # Since we are not retaining the ip we will use the random reserved ip
+                if (not actual_ip):
+                    alias_ip['ipCidrRange'] = get_ip(
+                        compute, project, alias_ip_name,
+                        get_region_from_zone(zone)) + "/32"
+                i = i + 1
+        config['networkInterfaces'][0]['aliasIpRanges'] = alias_ip_ranges
+
+    return compute.instances().insert(project=project,
+                                      zone=zone,
+                                      body=config,
+                                      sourceMachineImage='projects/' + project +
+                                      '/global/machineImages/' +
+                                      name).execute()
+
+
+def wait_for_instance(compute, project, zone, name):
+    """
+    Function to wait for creation of machine image
+    """
+    logging.info('Waiting for VM to start...')
+    while True:
+        result = compute.instances().get(project=project,
+                                         zone=zone,
+                                         instance=name).execute()
+        if result['status'] == 'RUNNING':
+            # logging.info("VM", name, "created and running")
+            if 'error' in result:
+                raise Exception(result['error'])
+            return result
+
+        time.sleep(10)
+
+
+def create(project,
+           target_zone,
+           network,
+           subnet,
+           instance_name,
+           alias_ip_ranges,
+           node_group,
+           disk_names,
+           ip=None,
+           machine_type=None,
+           wait=True):
+    """
+    Main functin to create the instance
+    """
+    try:
+        waited_time = rate_limit.wait()  # wait before starting the task
+        logging.info(f"  task: waited for {waited_time} secs")
+        compute = get_compute()
+        logging.info('Creating instance %s' % (instance_name))
+
+        create_instance(compute, project, target_zone, network, subnet,
+                        instance_name, alias_ip_ranges, node_group,
+                        disk_names, ip, machine_type)
+        if wait:
+            wait_for_instance(compute, project, target_zone, instance_name)
+        logging.info(
+            'Instance %s created and running form source MachineImage %s' %
+            (instance_name, instance_name))
+        return instance_name
+
+    except Exception as ex:
```

Comment:
`Exception` here is too broad. I would rather use a custom exception and `sys.exit` rather than re-raising the exception (and would let any other exception unhandled).